analysis_driver/notification.py

```diff
@@ -1,4 +1,59 @@
-from egcg_core import notifications
+from egcg_core import notifications, clarity
+from pyclarity_lims.entities import Step
+from analysis_driver.exceptions import PipelineError
+
+
+class LimsNotification():
+    def __init__(self, name):
+        self.name = name
+        self.sample_list = clarity.get_list_of_samples(self.name)
+        self.artifacts = [sample.artifact for sample in self.sample_list]
+        self.step = None
+
+    @property
+    def lims(self):
+        return clarity.connection()
+
+    @property
+    def postseqlab_stages(self):
+        return self.lims.get_workflows(name='PostSeqLab EG 1.0 WF')[0].stages
+
+    def get_stage(self, stage_name):
```

Comment:
The stage is the same across the entire run and does not need to be searched for every time.